plugin/internal/remotecall/remotecall.go

```diff
@@ -0,0 +1,227 @@
+package remotecall
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const (
+	UserAgentKey        = "User-Agent"
+	UserAgentSSHGateway = "SSH-gateway"
+	AcceptKey           = "Accept"
+	ContentType         = "Content-Type"
+	ApplicationJson     = "application/json"
+
+	AuthTokenUserClusterMapping = "authToken"
+	AuthKeyAuthenticator        = "Authorization"
```

Comment:
the namings here feel quite weird TBH - e.g. `AuthKeyAuthenticator` for sth with value `Authorization`, same for authtoken, why the suffix of clustermappinkg?